dll/win32/shell32/CShellLink.cpp

```diff
@@ -1768,27 +1774,8 @@ HRESULT STDMETHODCALLTYPE CShellLink::GetIconLocation(UINT uFlags, PWSTR pszIcon
 HRESULT STDMETHODCALLTYPE
 CShellLink::Extract(PCWSTR pszFile, UINT nIconIndex, HICON *phiconLarge, HICON *phiconSmall, UINT nIconSize)
 {
-    SHFILEINFOW info;
-
-    if (phiconLarge)
-    {
-        SHGetFileInfoW(pszFile, 0, &info, sizeof(info),
-                       SHGFI_ICON | SHGFI_LARGEICON | SHGFI_LINKOVERLAY);
-        *phiconLarge = info.hIcon;
-        if (!info.hIcon)
-            return E_FAIL;
-    }
-
-    if (phiconSmall)
-    {
-        SHGetFileInfoW(pszFile, 0, &info, sizeof(info),
-                       SHGFI_ICON | SHGFI_SMALLICON | SHGFI_LINKOVERLAY);
-        *phiconSmall = info.hIcon;
-        if (!info.hIcon)
-            return E_FAIL;
-    }
-
-    return S_OK;
+    UNIMPLEMENTED;
```

Comment:
Is this intentional?

Reply:
Could you explain why? It didn't actually work as intended?

Reply:
My test results say it's "unimplemented".

Reply:
I was wrong. It should be implemented. See ShellLinkTest.zip and ShellLinkTest.png of CORE-11407.

Reply:
Implemented. Done.